src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -1148,6 +1162,53 @@ class module_sram : public module_impl
     m_dirty = false;
   }
 
+  uint32_t*
+  fill_ert_aie2p(uint32_t *payload) const
+  {
+     if (m_preempt_save_bo && m_preempt_restore_bo) {
+       // ipu preemption
+       auto ipu = reinterpret_cast<ert_ipu_preempt_data*>(payload);
```

Comment:
Quote from email:
BTW, I also post a reference on how driver should make use of the new ERT command (not PREEMPTION though, working on it).

Do you mean this code path is not tested using SNL yet?

Reply:
On the contrary, the reference code was exactly used for testing purpose on SNL. We can not integrate hw_emu before merging XRT. That is why the hw_emu changes are just a reference for now.

Reply:
> On the contrary, the reference code was exactly used for testing purpose on SNL. We can not integrate hw_emu before merging XRT. That is why the hw_emu changes are just a reference for now.

oh, I may answered your question incorrectly, Yes, preemption code path has not been fully tested yet. But it is necessary to have the coreutil part ready so that drivers can start the development